### PR TITLE
:sparkles: Add target platforms to application archetypes

### PR DESCRIPTION
**Pull request by sjd78**
Resolves: #186

### CODE REVIEW

File: enhancements/archetype-target-platforms/README.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      "name": "Kubernetes",
+      "targets": [
+        "konveyor.io/target=cloud-readiness",
+        "konveyor.io/target=openjdk21",
+        "konveyor.io/target=eap8"
+      ]
+    },
+    {
+      "name": "Traditional Standalone",
+      "targets": [
```
Comment (shawn-hurley) [inline enhancements/archetype-target-platforms/README.md:148]: For future proof, can we think about naming these rulesets or labels?

**Review by shawn-hurley — APPROVED** [review]
Just a small naming concern, just thinking about the future of more and more rulesets with more and more labels.

File: enhancements/archetype-target-platforms/README.md (hunk 1 above)
Comment (jortel) [reply enhancements/archetype-target-platforms/README.md:148]: We already have a first-class, user facing concept (resource) of a `Target` which is just a named collection of labels.  The goal seems to be to pre-select one or more of these targets when in the analysis wizard using matched archetypes. The PR/issue suggests the selection could be locked as well as a constraint on the user.  Given this, it seems like _targets_ should be a list of references (id) to existing `Target` resources.

Comment (jortel) [reply enhancements/archetype-target-platforms/README.md:148]: After discussion with Scott whereby users want to build the target platform with specific versions of targets, I think we need to define this with labels.  Given this, I would recommend that "labels" would be better than "targets".

**Review by jortel — APPROVED** [review]
LGTM

File: enhancements/archetype-target-platforms/README.md (hunk 1 above)
Comment (sjd78) [reply enhancements/archetype-target-platforms/README.md]: I changed the sample json prop to "labels".  I also updated the language in the rest of the enhancement to explicitly say "target labels".  This should make it more obvious the intent is to use the specific label within a target in the same way the analysis wizard shows all the targets as cards with a dropdown to select the label, but then only sends the specific labels to start the analysis.

**Pull request merged**